packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -77,21 +77,30 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
           automaticallyImplyLeading: !_isProductLoaded),
       floatingActionButton: FloatingActionButton.extended(
         onPressed: () async {
-          final LocalDatabase localDatabase = context.read<LocalDatabase>();
-          final DaoProduct daoProduct = DaoProduct(localDatabase);
-          final Product? localProduct = await daoProduct.get(widget.barcode);
-          if (localProduct == null) {
-            product = Product(
-              barcode: widget.barcode,
-            );
-            daoProduct.put(product);
-            localDatabase.notifyListeners();
-          }
-          localDatabase.upToDate.set(product);
-          if (mounted) {
+          if (_basicDetailsAdded ||
+              _nutritionFactsAdded ||
+              _uploadedImages.isNotEmpty) {
```

Comment:
Then we could get rid of `_isProductLoaded` everywhere, couldn't we? Anyway the name stinks.
Or make it a getter.

Reply:
LINE 77, though would it be better if i replace there by `_uploadedImages.isEmpty`

Reply:
Exactly. And remove `_isProductLoaded` as a variable as well.